lib/models/device.coffee

```diff
@@ -873,10 +874,9 @@ getDeviceModel = (deps, opts) ->
 			deviceTypes: configModel().getDeviceTypes()
 			application: applicationModel().get(applicationNameOrId, $select: [ 'id', 'device_type' ])
 		.then ({ application, device, deviceTypes }) ->
-			deviceDeviceType = find(deviceTypes, { slug: device.device_type })
-			appDeviceType = find(deviceTypes, { slug: application.device_type })
```

Comment:
This would throw an undefined exception if a user would try to move a device and either of the device type slugs are non-existing (the validation hook on the API was added at a later point of time).